### ISSUE TITLE
[Survival] icon does not change when using a frag bomb twice.

### ISSUE THREAD

**Issue by geraint425** [Hekili/hekili#1153]
**BEFORE SUBMITTING AN ISSUE:**

1.  build: v9.1.0-1.0.8a-beta1

2.  Check to see if someone else has already opened an issue report here.  Please don't submit repeat reports.

**Describe the Bug/Issue**
When using a shrapnel bomb, the icon does not change. Occurs when shrapnel bomb needs to be used 2 times.

**To Reproduce**
I start the fight.
I'm using Serpent Sting.
I am using shrapnel bomb.
After that, the icon does not change.

**Expected behavior**
When using the shrapnel bomb I want the icon to change to match the color of the shrapnel bomb. If the addon recommends using the shrapnel bomb 2 times then the color of the shrapnel bomb should change.

**Screenshots**
https://imgur.com/a/i820ogw

**WHEN SUBMITTING AN ISSUE:**

Please provide the following information.  Leaving it out may result in your ticket being ignored (because I won't have the information needed to explore your report).

1.  The information from the Issue Reporting tab ( /hekili > Issue Reporting ).  This provides your talent and gear information.  You can paste it to pastebin.com and provide a link here.

**https://pastebin.com/4Kfm3GM8**

2.  If reporting a disagreement with the addon's recommendations, please take a snapshot when you observe the issue and provide the snapshot.  This is not a screenshot.  A snapshot can be generated by binding the Pause feature on the Toggles tab.  When you press Pause, a snapshot of how the addon decided on its CURRENT recommendations is generated and saved to the Snapshots tab.  Copy and paste this to pastebin.com and provide the link here.
 
**https://pastebin.com/mPyYQEGD**

3.  If reporting an error (i.e., the addon freezes and/or disables itself), please install the BugSack and BugGrabber addons as they will enable you to paste the exact error message, with detail that will help me identify where the code error is located.  You can paste the error information here or use pastebin.com for that as well.

**PUT ERROR MESSAGE(S) HERE:**

**Comment by Hekili** [comment]
What am I supposed to be seeing in your screenshot?

**Comment by geraint425** [comment]
Well, it shows in the center on WA that at the moment I have a red shrapnel bomb and the addon shows a blue shrapnel bomb. Plus, you can also see that the first blue shrapnel bomb was used. Since the debuff from the blue shrapnel bomb hangs on the target, plus, again, WA shows in the center the buff from the blue shrapnel bomb. You have previously corrected this problem, She got out again.

**Comment by geraint425** [comment]
I saw that you made more changes to the addon. I installed them, but the problem remained. I suppose that maybe this is due to the fact that when we play through traps, the addon cannot understand whether the trap has flown or not. Here are the same conditions for the campaign. I assume that this problem cannot be fixed. For the addon, I see, starts its ability when they touch the target, and if such abilities as traps, bombs that are thrown from afar, the addon cannot communicate them correctly and therefore the rotation is not correct. So thanks for the trouble, but the Surv rotation is not playable.
